dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java

```diff
@@ -297,7 +294,6 @@ public void notificationCreated(RuntimeUpdateManager updateManager, RuntimeUpdat
         if (RuntimeUpdateNotification.CONFIG_UPDATES_DELIVERED.equals(notification.getName()) && jvmRestore.compareAndSet(true, false)) {
             debug(tc, () -> "Processing config on restore.");
             final CountDownLatch localLatch = new CountDownLatch(1);
```

Comment:
- There's no need for `localLatch`.  The latch is seeded with value `1` and is then used by `waitForConfig()` to block the checkpoint thread until the either the latch is decremented to `0`, times out, or the thread is interrupted,  You removed `waitForConfig()` -- i.e. you removed the wait on the latch -- so there's no need to call `localLatch.countDown()` in the `CompletionListener` methods. 
- I like that the `CompletionListener` methods provide trace that define the window of the restore config update.  If that's not a concern, then you could remove the `RuntimeUpdateListener` support from `CheckpointImpl`.

Reply:
Removed, thanks!

Reply:
Looks good.  Thank you @anjumfatima90.